core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala

```diff
@@ -86,27 +86,63 @@ object CDCReader extends CDCReaderImpl
   // CDC specific columns in data written by operations
   val CDC_COLUMNS_IN_DATA = Seq(CDC_PARTITION_COL, CDC_TYPE_COLUMN_NAME)
 
+  // A snapshot coupled with a schema mode that user specified
+  case class SnapshotWithSchemaMode(snapshot: Snapshot, schemaMode: DeltaBatchCDFSchemaMode)
+
   /**
    * A special BaseRelation wrapper for CDF reads.
    */
   case class DeltaCDFRelation(
-      schema: StructType,
+      snapshotWithSchemaMode: SnapshotWithSchemaMode,
       sqlContext: SQLContext,
-      deltaLog: DeltaLog,
       startingVersion: Option[Long],
       endingVersion: Option[Long]) extends BaseRelation with PrunedFilteredScan {
 
+    private val deltaLog = snapshotWithSchemaMode.snapshot.deltaLog
+
+    private lazy val latestVersionOfTableDuringAnalysis: Long = deltaLog.update().version
+
+    /**
+     * There may be a slight divergence here in terms of what schema is in the latest data vs what
+     * schema we have captured during analysis, but this is an inherent limitation of Spark.
+     *
+     * However, if there are schema changes between analysis and execution, since we froze this
+     * schema, our schema incompatibility checks will kick in during the scan so we will always
+     * be safe - Although it is a notable caveat that user should be aware of because the CDC query
```

Comment:
are users aware of this? should we update any documentation to make this clearer?

Reply:
Nothing is documented yet for this behavior, but we probably should cc @jose-torres